### ISSUE TITLE
python train.py -c configs/config.json -m 44k 主模型训练到60000次之后无法再继续训练？是什么原因？

### ISSUE THREAD

**Issue by TzyTman** [SUC-DriverOld/so-vits-svc-Chinese-Detaild-Documents#18]
python train.py -c configs/config.json -m 44k
INFO:44k:{'train': {'log_interval': 600, 'eval_interval': 2400, 'seed': 1234, 'epochs': 10000, 'learning_rate': 0.0001, 'betas': [0.8, 0.99], 'eps': 1e-09, 'batch_size': 25, 'fp16_run': False, 'half_type': 'fp16', 'lr_decay': 0.999875, 'segment_size': 10240, 'init_lr_ratio': 1, 'warmup_epochs': 0, 'c_mel': 45, 'c_kl': 1.0, 'use_sr': True, 'max_speclen': 512, 'port': '8001', 'keep_ckpts': 0, 'all_in_mem': False, 'vol_aug': True}, 'data': {'training_files': 'filelists/train.txt', 'validation_files': 'filelists/val.txt', 'max_wav_value': 32768.0, 'sampling_rate': 44100, 'filter_length': 2048, 'hop_length': 512, 'win_length': 2048, 'n_mel_channels': 80, 'mel_fmin': 0.0, 'mel_fmax': 22050, 'unit_interpolate_mode': 'nearest'}, 'model': {'inter_channels': 192, 'hidden_channels': 192, 'filter_channels': 768, 'n_heads': 2, 'n_layers': 6, 'kernel_size': 3, 'p_dropout': 0.1, 'resblock': '1', 'resblock_kernel_sizes': [3, 7, 11], 'resblock_dilation_sizes': [[1, 3, 5], [1, 3, 5], [1, 3, 5]], 'upsample_rates': [8, 8, 2, 2, 2], 'upsample_initial_channel': 512, 'upsample_kernel_sizes': [16, 16, 4, 4, 4], 'n_layers_q': 3, 'n_layers_trans_flow': 3, 'n_flow_layer': 4, 'use_spectral_norm': False, 'gin_channels': 768, 'ssl_dim': 768, 'n_speakers': 1, 'vocoder_name': 'nsf-hifigan', 'speech_encoder': 'vec768l12', 'speaker_embedding': False, 'vol_embedding': True, 'use_depthwise_conv': False, 'flow_share_parameter': False, 'use_automatic_f0_prediction': True, 'use_transformer_flow': False}, 'spk': {'Dailin': 0}, 'model_dir': './logs/44k'}
WARNING:44k:/home/mk/so-vits-svc-Dailin is not a git repository, therefore hash value comparison will be ignored.
./logs/44k/G_60000.pth
./logs/44k/G_60000.pth
load 
INFO:44k:Loaded checkpoint './logs/44k/G_60000.pth' (iteration 10000)
./logs/44k/D_60000.pth
load 
./logs/44k/D_60000.pth
load 
INFO:44k:Loaded checkpoint './logs/44k/D_60000.pth' (iteration 10000)
./logs/44k/D_60000.pth
load 
./logs/44k/D_60000.pth
/home/mk/anaconda3/envs/so-vits-svc/lib/python3.8/site-packages/torch/autograd/__init__.py:200: UserWarning: Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 1, 4], strides() = [4, 1, 1]
bucket_view.sizes() = [32, 1, 4], strides() = [4, 4, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
/home/mk/anaconda3/envs/so-vits-svc/lib/python3.8/site-packages/torch/autograd/__init__.py:200: UserWarning: Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 1, 4], strides() = [4, 1, 1]
bucket_view.sizes() = [32, 1, 4], strides() = [4, 4, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
INFO:44k:====> Epoch: 10000, cost 30.20 s

**Comment by SUC-DriverOld** [comment]
epoch满10000了，会自动停止，你可以通过修改config.json中的“epoch”字段的对应数值来扩大限制。但一般没必要，满10000epoch就可以差不多收炉了。

造成此现象两种可能的原因：
1. 训练集数量太少
2. 训练epoch已达开发者建议的最多epoch，没必要继续训练（除非你觉得效果不好就自己改配置文件继续训）

**Comment by TzyTman** [comment]
好的谢谢



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: Sucial ***@***.***&gt;
发送时间: 2024年4月19日 16:06
收件人: SUC-DriverOld/so-vits-svc-Deployment-Documents ***@***.***&gt;
抄送: tuzhengyao ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [SUC-DriverOld/so-vits-svc-Deployment-Documents] python train.py -c configs/config.json -m 44k 主模型训练到60000次之后无法再继续训练？是什么原因？ (Issue #18)





 
epoch满10000了，会自动停止，你可以通过修改config.json字段中的“epoch”对应数值来扩大限制。但一般没必要，满10000epoch就可以差不多收炉了
